### PR TITLE
Fix a few bugs relating to terminated connections

### PR DESCRIPTION
**Pull request by pwaller**
I was comparing the performance of hookbot to @JohannesEbke's pub/sub mechanism
and hit a few small bugs.

### CODE REVIEW

File: pkg/hookbot/hookbot.go [hunk 1]
```diff
@@ -308,3 +308,19 @@ func (h *Hookbot) ServeSubscribe(conn *websocket.Conn, r *http.Request) {
 		}
 	}
 }
+
+func IsConnectionClose(err error) bool {
```
Comment (drj11) [inline pkg/hookbot/hookbot.go]: blech

Comment (pwaller) [inline pkg/hookbot/hookbot.go]: I've done this tens of times now. This is the least horrible solution that reliably works.

Comment (pwaller) [inline pkg/hookbot/hookbot.go]: (For detecting this case in particular)

**Comment by pwaller** [comment]
Also, this branch isn't yet done.

File: pkg/hookbot/hookbot.go (hunk 1 above)
Comment (drj11) [inline pkg/hookbot/hookbot.go]: I wasn't bleching at your code; I was bleching at Go for making this the least horrible solution that reliably works.

Comment (pwaller) [inline pkg/hookbot/hookbot.go]: [Yes, well](http://bravenewgeek.com/go-is-unapologetically-flaawed-heres-why-we-use-it/). :dancer:

**Pull request merged**